pkg/R/RDD.R

```diff
@@ -1063,7 +1078,7 @@ setMethod("saveAsObjectFile",
           function(rdd, path) {
             # If the RDD is in string format, need to serialize it before saving it because when
             # objectFile() is invoked to load the saved file, only serialized format is assumed.
-            if (!rdd@env$serialized) {
+            if (rdd@env$serialized == "string") {
```

Comment:
So I think the real question is how can we programmatically tell `ObjectFile` whether to use `byte` or `row` if the file itself is already serialized? I'm considering putting the `byte` RDD constructor inside a `try` function and then calling `RDD(jrdd, "row")` if it fails, but that feels sort of clumsy.